index.js

```diff
@@ -130,122 +134,120 @@ function serializeResponse (requestId, protocolData) {
   const writer = new Writer()
   writeProtocolData(writer, protocolData)
 
-  return writeEnvelope(TYPE_RESPONSE, requestId, writer.getBuffer())
+  return writeCommonLedgerProtocolPacket(TYPE_RESPONSE, requestId, writer.getBuffer())
 }
 
 function deserializeResponse (buffer) {
-  const { type, requestId, contents } = readEnvelope(buffer)
-  const reader = new Reader(contents)
+  const { type, requestId, data } = readCommonLedgerProtocolPacket(buffer)
+  const reader = new Reader(data)
 
   const protocolData = readProtocolData(reader)
   return { requestId, protocolData }
 }
 
-function serializeError ({ ilp }, requestId, protocolData) {
+function serializeError (error, requestId, protocolData) {
   const writer = new Writer()
-  const packet = Buffer.from(ilp, 'base64')
-
-  writer.write(packet)
+  const ilpPacket = serializeIlpError(error)
```

Comment:
This should check whether the error is a buffer or an object, and just serialize it if it's an error. That way if the error was already serialized (returned by some other function or something) it can be passed in here.